presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java

```diff
@@ -149,18 +150,28 @@ public Page getOutput()
 
         QueryInfo queryInfo = queryPerformanceFetcher.getQueryInfo(operatorContext.getDriverContext().getTaskId().getQueryId());
         checkState(queryInfo.getOutputStage().isPresent(), "Output stage is missing");
-        checkState(queryInfo.getOutputStage().get().getSubStages().size() == 1, "Expected one sub stage of explain node");
 
         if (!hasFinalStageInfo(queryInfo.getOutputStage().get())) {
             return null;
         }
         String plan;
         switch (format) {
             case TEXT:
-                plan = textDistributedPlan(queryInfo.getOutputStage().get().getSubStages().get(0), functionAndTypeManager, operatorContext.getSession(), verbose);
+                StringBuilder planStringBuilder = new StringBuilder();
+                for (StageInfo stageInfo : queryInfo.getOutputStage().get().getSubStages()) {
+                    planStringBuilder.append("Stage ID: ").append(stageInfo.getStageId().toString()).append('\n')
+                            .append(textDistributedPlan(stageInfo, functionAndTypeManager, operatorContext.getSession(),
+                                    verbose));
+                }
+                plan = planStringBuilder.toString();
                 break;
             case JSON:
-                plan = jsonDistributedPlan(queryInfo.getOutputStage().get().getSubStages().get(0), functionAndTypeManager, operatorContext.getSession());
+                StringJoiner planStringJoiner = new StringJoiner(",", "[", "]");
```

Comment:
Since the output needs to be valid JSON, we should probably use a JSON codec for lists and render it through that instead. It likely will require some more refactoring of the `PlanPrinter` class to achieve that. Joining strings like this is prone to errors/generating invalid JSON